### PR TITLE
Removed aeson dependency

### PR DESCRIPTION
**Pull request by j-hui**
Addressing #10

### CODE REVIEW

**Comment by j-hui** [comment]
When testing, I encountered this error:

```
Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for disjoint-containers-0.2.4:
    base-4.9.1.0 from stack configuration does not match >=4.11.1 && <5  (latest matching version is 4.12.0.0)
needed since disjoint-containers is a build target.

Some different approaches to resolving this:

  * Set 'allow-newer: true' in /Users/j-hui/.stack/config.yaml to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to attempt to find some working build configuration. This can be convenient when dealing with many complicated constraint
    errors, but results may be unpredictable.

  * Build requires unattainable version of base. Since base is a part of GHC, you most likely need to use a different GHC version with the matching base.

Plan construction failed.
```

@andrewthad I'm still somewhat new to using the Stack build system, but this appears to have been caused by using the `lts-9.11` resolver, which only provides `base-4.9.1.0` (https://www.stackage.org/diff/lts-9.11/lts-12.3). `lts-12.3` appears to be the earliest resolver with a compatible version of the `base` package.

**Comment by chessai** [comment]
Overall changes look good, but i have a question and a comment.

Comment: writing that we need the semigroup instance in order to define the monoid instance is redundant since that follows from the definition of a superclass

Question: why put the JSON instances in the test suite at all? They're no longer part of the package itself, so it's like testing part of the test suite. If we need the instances back we can get them from older commits

File: stack.yaml [hunk 1]
```diff
@@ -1,8 +1,12 @@
-resolver: lts-9.11
+resolver: lts-12.3
 packages:
 - .
 extra-deps:
-- 'quickcheck-classes-0.1'
+# - 'quickcheck-classes-0.1'
```
Comment (andrewthad) [inline stack.yaml]: Just delete this line instead of commenting it out

File: src/Data/DisjointMap.hs [hunk 2]
```diff
@@ -395,6 +384,7 @@ name as they would call it. We begin with a rating data type:
 
 >>> import Data.Function ((&))
 >>> data Rating = Lowest | Low | Medium | High | Highest deriving (Eq,Ord,Show)
+>>> instance Semigroup Rating where (<>) = min -- needed for Monoid Rating
```
Comment (andrewthad) [inline src/Data/DisjointMap.hs]: The comment "needed for Monoid Rating" does not provide any information since `Semigroup` is a superclass of `Monoid`.

**Review by andrewthad — CHANGES_REQUESTED** [review]
This looks pretty good. Like @chessai suggested, just remove the json stuff from the test suite. It can be tested in another package if necessary.

File: stack.yaml (hunk 1 above)
Comment (j-hui) [reply stack.yaml]: oops, my bad. done

**Comment by j-hui** [comment]
@andrewthad  done!

**Comment by andrewthad** [comment]
This looks good to me. @chessai  do you have any other concerns?

**Comment by chessai** [comment]
No

**Comment by chessai** [comment]
Thanks @j-hui

**Pull request merged**